### PR TITLE
refactor!: use gmt class to generalize cli DES generation

### PR DESCRIPTION
**Pull request by macintoshpie**
#### Any background context you want to provide?
GMT CLI currently only supports DES with timeseries loads.

#### What does this PR accomplish?
Allows CLI to generate packages with any load types (e.g. timeseries, spawn, teaser)

#### How should this be manually tested?
Create a geojson and sys param with one or more loads (at least one non-timeseries load), and run
```
poetry run uo_des create-model <sys param path> <geojson path> <output dir>
```
Verify the package looks correct (most importantly the correct load types were used). Try and simulate in Dymola.

#### What are the relevant tickets?
#370

### CODE REVIEW

File: geojson_modelica_translator/geojson_modelica_translator.py [hunk 1]
```diff
@@ -37,147 +37,166 @@
...
+    :param sys_params: SystemParameters
+    :return: list[Coupling], list of couplings to be passed to CouplingGraph
+    """
+    # Current implementation assumes that all generated district energy system models will have:
+    #   - one heating plant
+    #   - one cooling plant
+    #   - one heating distribution network
+    #   - one cooling distribution network
+    #   - one heating and cooling ETS per load
+    # NOTE: loads can be of any type/combination
```
Comment (macintoshpie) [inline geojson_modelica_translator/geojson_modelica_translator.py:94]: Important note. Maybe GMT will support my customizable configuration of plants/etses/etc down the road and we will no longer be able to make this assumption

File: geojson_modelica_translator/geojson_modelica_translator.py [hunk 2]
```diff
@@ -37,147 +37,166 @@
...
+            raise FileNotFoundError(f'GeoJSON file path does not exist: {geojson_filepath}')
+        if not Path(sys_params_filepath).exists():
+            raise FileNotFoundError(f'System parameters file path does not exist: {sys_params_filepath}')
+
+        self._system_parameters = SystemParameters(sys_params_filepath)
+
+        geojson_ids = self._system_parameters.get_default(
+            '$.buildings.custom[*].geojson_id',
+            []
+        )
```
Comment (macintoshpie) [inline geojson_modelica_translator/geojson_modelica_translator.py:175]: the loads declared in sys params determine which loads are parsed from geojson

File: geojson_modelica_translator/model_connectors/plants/cooling_plant.py [hunk 3]
```diff
@@ -49,10 +49,9 @@
 class CoolingPlant(PlantBase):
     model_name = 'CoolingPlant'
 
-    def __init__(self, system_parameters, template_name='CentralCoolingPlant'):
```
Comment (macintoshpie) [inline geojson_modelica_translator/model_connectors/plants/cooling_plant.py:52]: This isn't used, I don't think we need this.

File: management/uo_des.py [hunk 4]
```diff
@@ -148,118 +131,48 @@ def build_sys_param(model_type, sys_param_filename, scenario_file, feature_file,
     type=click.Path(exists=True, file_okay=True, dir_okay=False),
 )
 @click.argument(
-    "project_name",
+    "project_path",
```
Comment (macintoshpie) [inline management/uo_des.py:134]: "breaks" cli api, but I think this is a more appropriate name

File: geojson_modelica_translator/geojson/urbanopt_geojson.py [hunk 5]
```diff
@@ -73,41 +75,40 @@ class UrbanOptGeoJson(object):
...
-        for b in self.buildings:
-            val_res = self.schemas.validate("building", b.feature.properties)
-            if len(val_res) > 0:
-                status = False
-                res = {"id": b.feature.properties["id"], "errors": val_res}
-                validations["building"].append(res)
-
-        return status, validations
+        for feature in self.data.features:
+            if feature["properties"]["type"] == "Building":
```
Comment (nllong) [inline geojson_modelica_translator/geojson/urbanopt_geojson.py:95]: We've had this stub in here for quite some time. We should revisit this sometime in the future and clean up as needed.

File: geojson_modelica_translator/geojson_modelica_translator.py [hunk 6]
```diff
@@ -37,147 +37,166 @@
 """
 
 import logging
-import os
+from pathlib import Path
```
Comment (nllong) [inline geojson_modelica_translator/geojson_modelica_translator.py:40]: Nate has officially converted us :]

File: geojson_modelica_translator/geojson_modelica_translator.py (hunk 2 above)
Comment (nllong) [reply geojson_modelica_translator/geojson_modelica_translator.py:175]: most excellent

File: geojson_modelica_translator/model_connectors/plants/cooling_plant.py [hunk 7]
```diff
@@ -68,6 +67,10 @@ def to_modelica(self, scaffold):
         mos_wet_bulb_filename = self.system_parameters.get_param(
             "$.district_system.default.central_cooling_plant_parameters.mos_wet_bulb_filename"
         )
+
+        if mos_wet_bulb_filename is None:
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/plants/cooling_plant.py:71]: nice catch

File: management/uo_des.py (hunk 4 above)
Comment (nllong) [reply management/uo_des.py:134]: I don't think we document the CLI as `uo_des --project_name` so we should be good on the documentation side... hopefully.

File: tests/management/test_uo_des.py [hunk 8]
```diff
@@ -40,17 +43,39 @@ def test_cli_builds_sys_params(self):
         assert (self.output_dir / 'test_sys_param.json').exists()
 
     def test_cli_makes_model(self):
-        if (self.output_dir / 'modelica_project').exists():
-            rmtree(self.output_dir / 'modelica_project')
+        # WARNING: This test assumes test_cli_builds_sys_params has already run
```
Comment (nllong) [inline tests/management/test_uo_des.py:46]: 😬

File: tests/model_connectors/test_time_series_mft_ets.py [hunk 9]
```diff
@@ -41,8 +41,8 @@
 
 import pytest
 from buildingspy.io.outputfile import Reader
-from geojson_modelica_translator.geojson_modelica_translator import (
-    GeoJsonModelicaTranslator
+from geojson_modelica_translator.geojson.urbanopt_geojson import (
+    UrbanOptGeoJson
```
Comment (nllong) [inline tests/model_connectors/test_time_series_mft_ets.py:45]: I like this cleanup!

**Review by nllong — APPROVED** [review]
Well done! This is a comprehensive and impactful update!

**Pull request merged**